src/liballoc/string.rs

```diff
@@ -378,6 +378,7 @@ impl String {
     /// ```
     /// let s = String::new();
     /// ```
+    #[must_use]
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn new() -> String {
```

Comment:
This one's another case of a trivial method where, sure, not using it isn't ideal stylistically, but it's just setting 3 pointers to constants, so overall I feel "meh".

Reply:
Ok, for now I removed #[must_use] from std::String::new()